services/web/server/src/simcore_service_webserver/projects/projects_db.py

```diff
@@ -571,11 +333,154 @@ async def get_project(
+                    # pylint: disable=no-value-for-parameter
+                    projects.update()
+                    .values(**convert_to_db_names(new_project_data))
+                    .where(projects.c.id == current_project[projects.c.id.key])
+                    .returning(literal_column("*"))
+                )
+                project = await result.fetchone()
+                assert project  # nosec
+                await self.upsert_project_linked_product(
+                    ProjectID(project_uuid), product_name, conn=conn
+                )
+                log.debug(
+                    "DB updated returned row project=%s",
+                    json_dumps(dict(project.items())),
+                )
+                user_email = await self._get_user_email(conn, project.prj_owner)
+
+                tags = await self._get_tags_by_project(
+                    conn, project_id=project[projects.c.id]
+                )
+                return convert_to_schema_names(project, user_email, tags=tags)
+
+    async def update_project_without_checking_permissions(
+        self,
+        project_data: dict,
+        project_uuid: ProjectIDStr,
+        *,
+        hidden: Optional[bool] = None,
+    ) -> bool:
+        """The garbage collector needs to alter the row without passing through the
+        permissions layer."""
+        async with self.engine.acquire() as conn:
+            # update timestamps
+            project_data["lastChangeDate"] = now_str()
+            # now update it
+            updated_values = convert_to_db_names(project_data)
+            if hidden is not None:
+                updated_values["hidden"] = hidden
+            result = await conn.execute(
+                # pylint: disable=no-value-for-parameter
+                projects.update()
+                .values(**updated_values)
+                .where(projects.c.uuid == project_uuid)
+            )
+            return result.rowcount == 1
+
+    async def delete_project(self, user_id: int, project_uuid: str):
+        log.info(
+            "Deleting project with %s for user with %s",
+            f"{project_uuid=}",
+            f"{user_id}",
+        )
+
+        async with self.engine.acquire() as conn:
+            async with conn.begin() as _transaction:
+                project = await self._get_project(
+                    conn, user_id, project_uuid, for_update=True
+                )
+                # if we have delete access we delete the project
+                user_groups: list[RowProxy] = await self._list_user_groups(
+                    conn, user_id
+                )
+                check_project_permissions(project, user_id, user_groups, "delete")
+                await conn.execute(
+                    # pylint: disable=no-value-for-parameter
+                    projects.delete().where(projects.c.uuid == project_uuid)
+                )
+
+    #
+    # Project WORKBENCH / NODES
+    #
+
+    async def patch_project_workbench(
```

Comment:
maybe then rename to ```update_project_workbench```
minor: and btw not sure it is not still a CRUD operation.

Reply:
it is indeed a CRUD but i wanted to differentiate operations in the main resource "project" from those in the sub-resources "project/workbench".
I can call it update yes.